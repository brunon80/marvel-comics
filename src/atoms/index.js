export { default as TextInput } from './inputs/TextInput'
export { default as Wrapper } from './wrapper/Wrapper'