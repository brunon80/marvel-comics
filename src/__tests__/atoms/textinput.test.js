import { render, screen, fireEvent } from '@testing-library/react'
import { TextInput } from '../../atoms'

describe('TextInput tests', () => {
  const setup = () => {
    const utils = render(<TextInput aria-label="search" />)
    const input = utils.getByLabelText('search')
    return {
      input,
      ...utils,
    }
  }

  it('should render correctly', () => {
    setup()
    expect(screen.getByLabelText('search')).toBeInTheDocument()
  })

  it('should write on text input', () => {
    const value = 'deadpool'
    const { input } = setup()

    fireEvent.change(input, { target: { value } })
    expect(input.value).toBe(value)
  })

  it('should macth snapshot <TextInput />', async () => {
    const { input } = setup()
    expect(input).toMatchSnapshot()
  })
})