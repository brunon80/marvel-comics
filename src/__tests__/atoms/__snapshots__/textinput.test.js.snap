// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput tests should macth snapshot <TextInput /> 1`] = `
<input
  aria-label="search"
  type="text"
/>
`;
